File: endpoints/src/main/java/com/cromoteca/samples/endpoints/EnumsEndpoint.java
package com.cromoteca.samples.endpoints;

import com.vaadin.flow.server.auth.AnonymousAllowed;
import com.vaadin.hilla.Endpoint;

@Endpoint
@AnonymousAllowed
public class EnumsEndpoint {
  public Color getRed() {
    return Color.RED;
  }

  public static enum Color {
    RED,
    GREEN,
    BLUE
  }
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/EnumsEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Color from './EnumsEndpoint/Color.js';
import type { EndpointRequestInit } from '@hilla/frontend';

async function getRed(init?: EndpointRequestInit): Promise<Color> {
    return client.call('com.cromoteca.samples.endpoints.EnumsEndpoint', 'getRed', {}, init);
}

const EnumsEndpoint = {
    getRed,
};

export default EnumsEndpoint;
